src/Build/Telemetry/ITelemetryCollector.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Build.BackEnd.Logging;
+
+namespace Microsoft.Build.Telemetry;
+
+/// <summary>
+/// A build component responsible for accumulating telemetry data from worker node and then sending it to main node
+/// at the end of the build.
+/// </summary>
+internal interface ITelemetryCollector
```

Comment:
I would prefer different terminology than Collector to prevent confusion with OTel and VS OTel concept of a collector.

Reply:
quick suggestions from GPT I like:
- **NodeTelemetrySender**
- **NodeTelemetryReporter**
- **TelemetryForwarder**
- **TelemetryAccumulator**